src/hotspot/share/opto/parseHelper.cpp

```diff
@@ -680,3 +679,39 @@ void PEAState::validate() const {
   });
 }
 #endif
+
+MergeProcessor::MergeProcessor(PEAState& target) : _state(target) {
+}
+void MergeProcessor::merge(const PEAState& newin, GraphKit* kit, RegionNode* region, int pnum) {
+  // intersection of two aliases
+  Unique_Node_List alias1;
+  Unique_Node_List alias2;
+
+  _state.aliases()->iterate([&](Node* alias, ObjID obj) {
+                             alias1.push(alias);
+                             return true;
+                           });
+  newin.aliases()->iterate([&](Node* alias, ObjID obj) {
+                            alias2.push(alias);
+                            return true;
+                          });
+  VectorSet set = alias1.member_set();
+  set &= alias2.member_set();
+  alias1.remove_useless_nodes(set);
+
+  for (uint i = 0; i < alias1.size(); ++i) {
+    Node* var = alias1.at(i);
+    VirtualState* vs1 = _state.as_virtual(var);
+    VirtualState* vs2 = newin.as_virtual(var);
+    if (vs1 != nullptr && vs2 != nullptr) {
+      vs1->merge(vs2, kit, region, pnum);
+    }
+  }
+}
+
+MergeProcessor::~MergeProcessor() {
+
+}
+
+void MergeProcessor::process_phi(PhiNode* phi) {
```

Comment:
what is this function for?

Reply:
This is a placeholder function. I plan to use it in the loopbody of merge_common. 
yes, I better clean up everything before merging it.